src/nbl/ext/DebugDraw/CDrawAABB.cpp

```diff
@@ -0,0 +1,429 @@
+		std::make_pair(bool(creationParams.utilities), "Invalid `creationParams.utilities` is nullptr!"),
+		std::make_pair(bool(creationParams.transfer), "Invalid `creationParams.transfer` is nullptr!"),
+		std::make_pair(bool(creationParams.renderpass), "Invalid `creationParams.renderpass` is nullptr!"),
+		(creationParams.assetManager && creationParams.utilities && creationParams.transfer && creationParams.renderpass) ? std::make_pair(bool(creationParams.utilities->getLogicalDevice()->getPhysicalDevice()->getQueueFamilyProperties()[creationParams.transfer->getFamilyIndex()].queueFlags.hasFlags(IQueue::FAMILY_FLAGS::TRANSFER_BIT)), "Invalid `creationParams.transfer` is not capable of transfer operations!") : std::make_pair(false, "Pass valid required DrawAABB::S_CREATION_PARAMETERS!")
+		});
+
+	system::logger_opt_ptr logger = creationParams.utilities->getLogger();
+	for (const auto& [ok, error] : validation)
+		if (!ok)
+		{
+			logger.log(error, ILogger::ELL_ERROR);
+			return false;
+		}
+
+	return true;
+}
+
+smart_refctd_ptr<IGPUGraphicsPipeline> DrawAABB::createPipeline(SCreationParameters& params, const IGPUPipelineLayout* pipelineLayout, const std::string& vsPath, const std::string& fsPath)
+{
+	auto system = smart_refctd_ptr<ISystem>(params.assetManager->getSystem());
+	auto* set = params.assetManager->getCompilerSet();
+	auto compiler = set->getShaderCompiler(IShader::E_CONTENT_TYPE::ECT_HLSL);
+	auto includeFinder = make_smart_refctd_ptr<IShaderCompiler::CIncludeFinder>(smart_refctd_ptr(system));
+	auto includeLoader = includeFinder->getDefaultFileSystemLoader();
+	includeFinder->addSearchPath(NBL_ARCHIVE_ENTRY.data(), includeLoader);
+
+	auto compileShader = [&](const std::string& filePath, IShader::E_SHADER_STAGE stage) -> smart_refctd_ptr<IShader>
+		{
+			IAssetLoader::SAssetLoadParams lparams = {};
+			lparams.logger = params.utilities->getLogger();
+			lparams.workingDirectory = NBL_ARCHIVE_ENTRY.data();
+			auto bundle = params.assetManager->getAsset(filePath, lparams);
+			if (bundle.getContents().empty() || bundle.getAssetType() != IAsset::ET_SHADER)
+			{
+				params.utilities->getLogger()->log("Shader %s not found!", ILogger::ELL_ERROR, filePath.c_str());
+				exit(-1);
+			}
+
+			const auto assets = bundle.getContents();
+			assert(assets.size() == 1);
+			smart_refctd_ptr<IShader> shaderSrc = IAsset::castDown<IShader>(assets[0]);
+			if (!shaderSrc)
+				return nullptr;
+
+			CHLSLCompiler::SOptions options = {};
+			options.stage = stage;
+			options.preprocessorOptions.sourceIdentifier = filePath;
+			options.preprocessorOptions.logger = params.utilities->getLogger();
+			options.preprocessorOptions.includeFinder = includeFinder.get();
+			shaderSrc = compiler->compileToSPIRV((const char*)shaderSrc->getContent()->getPointer(), options);
+
+			return params.utilities->getLogicalDevice()->compileShader({ shaderSrc.get() });
```

Comment:
@AnastaZIuk can we make @YasInvolved use CMake to make precompiled SPIR-V with NSC right way in a future PR to CI the shaders?